scan

```diff
@@ -111,118 +109,77 @@ def run(options=None, cache_dir=None, results_dir=None):
             raise FileNotFoundError
     domains = scan_utils.handle_domains_argument(options["domains"], cache_dir)
 
-    # Split the scans into new-style OO scanners and old-style scan modules:
-    scanner_classes, scans = scan_utils.build_scan_lists(
-        options["scan"].split(","))
+    # Import the scanners:
+    scans = scan_utils.build_scanner_list(options["scan"].split(","))
 
-    # Which scanners to run the domain through.
-    if len(scans) > 0:
-        scan_domains(scans, domains, options)
-    if len(scanner_classes) > 0:
-        scanobject_domains(scanner_classes, domains, options)
+    # Now that we've loaded the modules, we can process args with them:
+    options, unknown = scan_utils.handle_scanner_arguments(scans, options, unknown)
+
+    # Kick off the scanning:
+    scan_domains(scans, domains, options)
 
 
 ###
-# Given the selected scanners, and input domains, run each domain
+# Given the selected scanner modules, and input domains, run each domain
 # through each scanner.
 #
 # Produces a CSV for each scan, with each domain and results.
 ###
-def scan_domains(scanners, domains, options):
+def scan_domains(scanners: List[ModuleType], domains: Path,
+                 options: dict) -> None:
     # Clear out existing result CSVs, to avoid inconsistent data.
     results_dir = options["_"]["results_dir"]
-    for result in glob.glob("%s/*.csv" % results_dir):
+    for result in Path(os.path.curdir, results_dir).glob("*.csv"):
         os.remove(result)
 
     # Store local errors/timing info, and if using Lambda, trigger the
     # Lambda post-processing pipeline to get Lambda timing/usage info.
     meta = options.get("meta", False)
-    # Generate a random UUID for the entire scan.
-    scan_uuid = str(uuid.uuid4())
 
     # Run through each scanner and open a file and CSV for each.
     handles = {}
+    durations = {}
+    scan_uuid = str(uuid.uuid4())
+    # Store scan UUID.
+    logging.debug("[%s] Scan UUID." % scan_uuid)
     for scanner in scanners:
         name = scanner.__name__.split(".")[-1]  # e.g. 'pshtt'
-        scanner_filename = "%s/%s.csv" % (results_dir, name)
```

Comment:
Noting that there was a regression here - this line was remove, and replaced in `scan_utils` with `scanner_csv_path = Path(results_dir, name).resolve()`, which was missing the `.csv` extension.